test/gleam/erlang/file_test.gleam

```diff
@@ -46,8 +55,38 @@ pub fn label_test() {
 
   assert Ok(Nil) = file.delete(path)
   assert Error(file.Enoent) = file.read(from: path)
+  delete_tmp_dir()
+}
+
+pub fn dir_test() {
+  make_tmp_dir()
+  let path = tmp_path("missing_dir/foo")
+  assert Error(file.Enoent) = file.make_dir(path: path)
+  assert False = file.is_dir(path: path)
+  assert False = file.is_file(path: path)
+
+  let path = tmp_path("bar")
+  assert Ok(Nil) = file.make_dir(path: path)
+  assert True = file.is_dir(path: path)
+  assert True = file.is_file(path: path)
+
+  let nested_path = tmp_path("bar/baz")
+  assert Ok(Nil) = file.make_dir(path: nested_path)
+  assert Ok(Nil) = file.delete_dir_r(path: path)
+  assert Error(file.Enoent) = file.delete_dir(path: path)
+
+  delete_tmp_dir()
 }
 
 fn tmp_path(filename: String) {
-  string.concat(["test/tmp/", filename])
+  string.concat([tmp_dir, filename])
+}
+
+fn make_tmp_dir() {
+  delete_tmp_dir()
+  file.make_dir(tmp_dir)
```

Comment:
Can we add assertions to these? It would be confusing if these were to silently fail.